### ISSUE TITLE
What's the actual acc for ESIM model?

### ISSUE THREAD

**Issue by bzqweiyi** [pengming617/text_matching#11]
用项目中的数据集进行训练，实际测试结果达不到论文的预期。

**Comment by pengming617** [comment]
里面有些参数需要调节下 比如l2_lamda,词向量维度等等 还有就是论文里面别人用的是预训练好的词向量 这些都有影响的

**Comment by bzqweiyi** [comment]
Hi 博主：

第20次迭代的损失为：0.59667087;准确率为：0.752356
176it [00:04, 40.98it/s]
分类报告:
               precision    recall  f1-score   support

           0       0.72      0.46      0.56      4400
           1       0.60      0.82      0.70      4400

    accuracy                           0.64      8800
   macro avg       0.66      0.64      0.63      8800
weighted avg       0.66      0.64      0.63      8800

验证集：loss 0.837242, acc 0.642386, f1 0.630396

这是我新的结果，原先采用不分词的方式，acc只有0.4左右；后来采用分词后，进行训练；其参数如下所示，改变了词向量的维度、隐藏层规模、l2、lr。

        self.embedding_size = 200  # 词向量维度
        self.hidden_num = 200  # 隐藏层规模
        self.l2_lambda = 0.0 # 0.01 验证集：loss 0.917116, acc 0.603864, f1 0.584642
        self.learning_rate = 0.002
        self.dropout_keep_prob = 0.5
        self.attn_size = 200
        self.K = 2

        self.epoch = 20
        self.Batch_Size = 50

准确率和你跑出的0.74还是有点差距的。
现在，我再尝试调整下参数，再新训练后模型的效果，是否有提升。

**Comment by bzqweiyi** [comment]
Hi 博主：

重新调试后，运行结果和参数设置如下：
第20次迭代的损失为：0.2162451;准确率为：0.9514075
68it [00:03, 21.26it/s]
分类报告:
               precision    recall  f1-score   support

           0       0.83      0.78      0.80      4355
           1       0.79      0.84      0.81      4349

    accuracy                           0.81      8704
   macro avg       0.81      0.81      0.81      8704
weighted avg       0.81      0.81      0.81      8704

验证集：loss 0.4786, acc 0.808824, f1 0.808689


    def __init__(self):
        self.embedding_size = 200  # 词向量维度
        self.hidden_num = 128  # 隐藏层规模
        self.l2_lambda = 0.0 # 0.01 验证集：loss 0.917116, acc 0.603864, f1 0.584642
        self.learning_rate = 0.001
        self.dropout_keep_prob = 0.7
        self.attn_size = 200
        self.K = 2

        self.epoch = 20
        self.Batch_Size = 128

其中，验证集上的loss0.4786，还是有点高，不过整体的准确率大幅提升。所有的数据是博主您git中提供。
现在有一个问题，我们项目中的语料，只有正例，没有反例，也就是标签都为1，没有为0的标签；
另外，能否提供大于两个句子的训练语料，也就是一句话，相同的说法有很多种，我们同时打完标签1，再进行训练，请博主指教！谢谢🙏！
万分感激！

祝好，
James

**Comment by pengming617** [comment]
模型还是有点过拟合的  看能不能再调整下  只有正例这个比较好解决 例如你的语料:A a 1
B b 1
C c 1
D d 1
E e 1
你可以自己去生成反例,例如A和B,b和D等都是0,随机生成就可以了


大于两个句子的训练语料中文的我没有  英文的应该有 DSSM paircnn_ranking论文里面应该有提供!




------------------ 原始邮件 ------------------
发件人: "james"<notifications@github.com>; 
发送时间: 2019年8月12日(星期一) 晚上8:54
收件人: "pengming617/text_matching"<text_matching@noreply.github.com>; 
抄送: "彭明"<549639484@qq.com>; "Comment"<comment@noreply.github.com>; 
主题: Re: [pengming617/text_matching] What's the actual acc for ESIM model?(#11)



Hi 博主：
 
 重新调试后，运行结果和参数设置如下：
 第20次迭代的损失为：0.2162451;准确率为：0.9514075
 68it [00:03, 21.26it/s]
 分类报告:
                precision    recall  f1-score   support
 
            0       0.83      0.78      0.80      4355
            1       0.79      0.84      0.81      4349
 
     accuracy                           0.81      8704
    macro avg       0.81      0.81      0.81      8704
 weighted avg       0.81      0.81      0.81      8704
 
 验证集：loss 0.4786, acc 0.808824, f1 0.808689
 
 
     def __init__(self):
         self.embedding_size = 200  # 词向量维度
         self.hidden_num = 128  # 隐藏层规模
         self.l2_lambda = 0.0 # 0.01 验证集：loss 0.917116, acc 0.603864, f1 0.584642
         self.learning_rate = 0.001
         self.dropout_keep_prob = 0.7
         self.attn_size = 200
         self.K = 2
 
         self.epoch = 20
         self.Batch_Size = 128
 
 其中，验证集上的loss0.4786，还是有点高，不过整体的准确率大幅提升。所有的数据是博主您git中提供。
 现在有一个问题，我们项目中的语料，只有正例，没有反例，也就是标签都为1，没有为0的标签；
 另外，能否提供大于两个句子的训练语料，也就是一句话，相同的说法有很多种，我们同时打完标签1，再进行训练，请博主指教！谢谢🙏！
 万分感激！
 
 祝好，
 James

**Comment by bzqweiyi** [comment]
感谢博主的提醒，原始我们语料组准备语料时，提供了一句话很多的正例，也就是相同的20个句子都是标签1，这个我的做法是否还是和您说的跟反例一样，也就是从20个正例中两两组合，同时打标签为1；或者有没有办法直接将这个20个相同的正例，直接喂给现在的模型呢？发自我的华为手机-------- 原始邮件 --------主题：Re: [pengming617/text_matching] What's the actual acc for ESIM model? (#11)发件人：pengming617 收件人：pengming617/text_matching 抄送：james ,Author 模型还是有点过拟合的  看能不能再调整下  只有正例这个比较好解决 例如你的语料:A a 1

B b 1

C c 1

D d 1

E e 1

你可以自己去生成反例,例如A和B,b和D等都是0,随机生成就可以了





大于两个句子的训练语料中文的我没有  英文的应该有 DSSM paircnn_ranking论文里面应该有提供!









------------------ 原始邮件 ------------------

发件人: "james"<notifications@github.com>; 

发送时间: 2019年8月12日(星期一) 晚上8:54

收件人: "pengming617/text_matching"<text_matching@noreply.github.com>; 

抄送: "彭明"<549639484@qq.com>; "Comment"<comment@noreply.github.com>; 

主题: Re: [pengming617/text_matching] What's the actual acc for ESIM model?(#11)







Hi 博主：

 

 重新调试后，运行结果和参数设置如下：

 第20次迭代的损失为：0.2162451;准确率为：0.9514075

 68it [00:03, 21.26it/s]

 分类报告:

                precision    recall  f1-score   support

 

            0       0.83      0.78      0.80      4355

            1       0.79      0.84      0.81      4349

 

     accuracy                           0.81      8704

    macro avg       0.81      0.81      0.81      8704

 weighted avg       0.81      0.81      0.81      8704

 

 验证集：loss 0.4786, acc 0.808824, f1 0.808689

 

 

     def __init__(self):

         self.embedding_size = 200  # 词向量维度

         self.hidden_num = 128  # 隐藏层规模

         self.l2_lambda = 0.0 # 0.01 验证集：loss 0.917116, acc 0.603864, f1 0.584642

         self.learning_rate = 0.001

         self.dropout_keep_prob = 0.7

         self.attn_size = 200

         self.K = 2

 

         self.epoch = 20

         self.Batch_Size = 128

 

 其中，验证集上的loss0.4786，还是有点高，不过整体的准确率大幅提升。所有的数据是博主您git中提供。

 现在有一个问题，我们项目中的语料，只有正例，没有反例，也就是标签都为1，没有为0的标签；

 另外，能否提供大于两个句子的训练语料，也就是一句话，相同的说法有很多种，我们同时打完标签1，再进行训练，请博主指教！谢谢🙏！

 万分感激！

 

 祝好，

 James

**Comment by pengming617** [comment]
现在的模型不支持这种  之前和你说的那两种模型可以支持的



------------------ 原始邮件 ------------------
发件人: "james"<notifications@github.com>; 
发送时间: 2019年8月12日(星期一) 晚上9:58
收件人: "pengming617/text_matching"<text_matching@noreply.github.com>; 
抄送: "彭明"<549639484@qq.com>; "Comment"<comment@noreply.github.com>; 
主题: Re: [pengming617/text_matching] What's the actual acc for ESIM model?(#11)



感谢博主的提醒，原始我们语料组准备语料时，提供了一句话很多的正例，也就是相同的20个句子都是标签1，这个我的做法是否还是和您说的跟反例一样，也就是从20个正例中两两组合，同时打标签为1；或者有没有办法直接将这个20个相同的正例，直接喂给现在的模型呢？发自我的华为手机-------- 原始邮件 --------主题：Re: [pengming617/text_matching] What's the actual acc for ESIM model? (#11)发件人：pengming617 收件人：pengming617/text_matching 抄送：james ,Author 模型还是有点过拟合的  看能不能再调整下  只有正例这个比较好解决 例如你的语料:A a 1 
 
 B b 1 
 
 C c 1 
 
 D d 1 
 
 E e 1 
 
 你可以自己去生成反例,例如A和B,b和D等都是0,随机生成就可以了 
 
 
 
 
 
 大于两个句子的训练语料中文的我没有  英文的应该有 DSSM paircnn_ranking论文里面应该有提供! 
 
 
 
 
 
 
 
 
 
 ------------------ 原始邮件 ------------------ 
 
 发件人: "james"<notifications@github.com>;  
 
 发送时间: 2019年8月12日(星期一) 晚上8:54 
 
 收件人: "pengming617/text_matching"<text_matching@noreply.github.com>;  
 
 抄送: "彭明"<549639484@qq.com>; "Comment"<comment@noreply.github.com>;  
 
 主题: Re: [pengming617/text_matching] What's the actual acc for ESIM model?(#11) 
 
 
 
 
 
 
 
 Hi 博主： 
 
   
 
  重新调试后，运行结果和参数设置如下： 
 
  第20次迭代的损失为：0.2162451;准确率为：0.9514075 
 
  68it [00:03, 21.26it/s] 
 
  分类报告: 
 
                 precision    recall  f1-score   support 
 
   
 
             0       0.83      0.78      0.80      4355 
 
             1       0.79      0.84      0.81      4349 
 
   
 
      accuracy                           0.81      8704 
 
     macro avg       0.81      0.81      0.81      8704 
 
  weighted avg       0.81      0.81      0.81      8704 
 
   
 
  验证集：loss 0.4786, acc 0.808824, f1 0.808689 
 
   
 
   
 
      def __init__(self): 
 
          self.embedding_size = 200  # 词向量维度 
 
          self.hidden_num = 128  # 隐藏层规模 
 
          self.l2_lambda = 0.0 # 0.01 验证集：loss 0.917116, acc 0.603864, f1 0.584642 
 
          self.learning_rate = 0.001 
 
          self.dropout_keep_prob = 0.7 
 
          self.attn_size = 200 
 
          self.K = 2 
 
   
 
          self.epoch = 20 
 
          self.Batch_Size = 128 
 
   
 
  其中，验证集上的loss0.4786，还是有点高，不过整体的准确率大幅提升。所有的数据是博主您git中提供。 
 
  现在有一个问题，我们项目中的语料，只有正例，没有反例，也就是标签都为1，没有为0的标签； 
 
  另外，能否提供大于两个句子的训练语料，也就是一句话，相同的说法有很多种，我们同时打完标签1，再进行训练，请博主指教！谢谢🙏！ 
 
  万分感激！ 
 
   
 
  祝好， 
 
  James
